scripts/azureml-assets/azureml/assets/release/model_publish_utils/mlflow_model_utils.py

```diff
@@ -0,0 +1,61 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+import logging
+import os
+import azureml.evaluate.mlflow as mlflow
+from mlflow.models import ModelSignature
+from sqlalchemy import true
+from transformers import  AutoTokenizer, AutoConfig, pipeline
+from transformers import (
+    AutoModelForSequenceClassification,
+    AutoModelForMaskedLM,
+    AutoModelForTokenClassification,
+    AutoModelForQuestionAnswering,
+    AutoModelWithLMHead
+)
+
+class MLFlowModelUtils:
+
+    def __init__(self, name, task_name, flavor, model_dir):
+        self.name = name
+        self.task_name = task_name
+        self.model_dir = model_dir
+        self.flavor = flavor
+
+    def _convert_to_mlflow_hftransformers(self):
+        config = AutoConfig.from_pretrained(self.name)
+        misc_conf = {"task_type": self.task_name}
+        task_model_mapping = {
+            "multiclass": AutoModelForSequenceClassification,
+            "multilabel": AutoModelForSequenceClassification,
+            "fill-mask": AutoModelForMaskedLM,
+            "ner": AutoModelForTokenClassification,
+            "question-answering": AutoModelForQuestionAnswering,
+            "summarization": AutoModelWithLMHead,
+            "text-generation": AutoModelWithLMHead,
+            "text-classification": AutoModelForSequenceClassification
+        }
+        if self.task_name in task_model_mapping:
+            model = task_model_mapping[self.task_name].from_pretrained(self.name, config=config)
+        elif "translation" in self.task_name:
+            model = AutoModelWithLMHead.from_pretrained(self.name, config=config)
+        else:
+            logging.error("Invalid Task Name")
+        tokenizer = AutoTokenizer.from_pretrained(self.name, config=config)
+        sign_dict = {"inputs": '[{"name": "input_string", "type": "string"}]', "outputs": '[{"type": "string"}]'}
+        if self.task_name == "question-answering":
+            sign_dict["inputs"] = '[{"name": "question", "type": "string"}, {"name": "context", "type": "string"}]'
+        signature = ModelSignature.from_dict(sign_dict)
+        self.mlflow_model_dir = self.name + "-mlflow"
```

Comment:
I name going to contain `/`? For eg. in case of `microsoft/deberta-base`.
Or are publishers expected to share valid model names? Should we add validations for spec info as well?

Reply:
when the name contains `/`, it only affects the az ml model create step. The error thrown when spec.yaml has / in the name is in accordance to the error one faces when locally creating a model using az cli. Since we are following az-cli schema for the spec.yaml file, the user should make sure that their spec file is inline with az cli requirements